packages/runtime-dom/src/patchProp.ts

```diff
@@ -93,9 +99,8 @@ function shouldSetAsProp(
     return false
   }
 
-  // #1787 form as an attribute must be a string, while it accepts an Element as
-  // a prop
-  if (key === 'form' && typeof value === 'string') {
+  // #1787, #2840
+  if (key === 'form' && isFormTag(el.tagName)) {
```

Comment:
I'm not sure this is good because we should still be able to do `:form="refToElement"`, no? I couldn't find the docs on the `form` attribute this time. If they are always readonly, then when do we need to check for the tag to be part of the form tag list?

Also, don't remove the comment, update it if it needs, it's easier while reading the code

Reply:
The form attribute only supports strings. The form DOMprop is always read-only on these elements.

There is no way to pass an actual HTMLElement as a reference to the form, only the form element's id selector.

Reply:
Thanks! Then do we need the check of the tag? I guess it's safer that way

Reply:
I'd say it's cleaner insofar as, if anyone wanted to set a `form` attribute on a non-form-related element (say a div) it would be set like any other attribute, not getting some special treatment it doesn't need.

I briefly thought about possible perf impact, but as this is a pretty special attribute (compared to `class` etc.), it's unlikely to have hundreds of these to check in a single patch run, even with a big form.